utils/validators/validators.go

```diff
@@ -0,0 +1,194 @@
+func ValidateEnvVars(envs []string) (envSettings EnvSettings, errs []error, warns []string) {
+	for _, env := range envs {
+		p := strings.SplitN(env, "=", 2)
+		if len(p) != 2 {
+			errs = append(errs, fmt.Errorf("failed to parse environment variable %q", env))
+			continue
+		}
+
+		var err error
+		k, v := strings.ToUpper(p[0]), strings.ToLower(p[1])
+		switch k {
+		case "PATH", "HOSTNAME", "TERM", "HOME", "PWD", "SHLVL", "_":
+			// skip default environment variables
+		case "DISABLE_UPDATES":
+			envSettings.DisableUpdates, err = strconv.ParseBool(v)
+			if err != nil {
+				err = fmt.Errorf("invalid value %q for environment variable %q", v, k)
+			}
+		case "DISABLE_TELEMETRY":
+			envSettings.DisableTelemetry, err = strconv.ParseBool(v)
+			if err != nil {
+				err = fmt.Errorf("invalid value %q for environment variable %q", v, k)
+			}
+		case "METRICS_RESOLUTION", "METRICS_RESOLUTION_HR":
+			if envSettings.MetricsResolutions.HR, err = validateStringMetricResolution(v); err != nil {
+				err = formatEnvVariableError(err, env, v)
+			}
+		case "METRICS_RESOLUTION_MR":
+			if envSettings.MetricsResolutions.MR, err = validateStringMetricResolution(v); err != nil {
+				err = formatEnvVariableError(err, env, v)
+			}
+		case "METRICS_RESOLUTION_LR":
+			if envSettings.MetricsResolutions.LR, err = validateStringMetricResolution(v); err != nil {
+				err = formatEnvVariableError(err, env, v)
+			}
+		case "DATA_RETENTION":
+			if envSettings.DataRetention, err = validateStringDataRetention(v); err != nil {
+				err = formatEnvVariableError(err, env, v)
+			}
+		default:
+			if !strings.HasPrefix(k, "GF_") {
+				warns = append(warns, fmt.Sprintf("unknown environment variable %q", env))
+			}
+		}
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return envSettings, errs, warns
+}
+
+func formatEnvVariableError(err error, env, value string) error {
+	switch e := err.(type) {
+	case InvalidDurationError:
+		return fmt.Errorf("environment variable %q has invalid duration %s", env, value)
+	case MinDurationError:
+		return fmt.Errorf("environment variable %q cannot be less then %s", env, e.Min)
+	case AliquotDurationError:
+		return fmt.Errorf("environment variable %q should be a multiple of %s", env, e.MultipleOf)
+	default:
+		return errors.Wrap(e, "unknown error")
+	}
+}
+
+// ValidateStringDuration validate duration as string value.
+func validateStringDuration(value string, min, multipleOf time.Duration) (time.Duration, error) {
+	d, err := time.ParseDuration(value)
+	if err != nil {
+		return d, InvalidDurationError("invalid duration error")
+	}
+
+	return validateDuration(d, min, multipleOf)
+}
+
+// ValidateDuration validate duration.
+func validateDuration(d, min, multipleOf time.Duration) (time.Duration, error) {
+	if d < min {
+		return d, MinDurationError{"min duration error", min}
+	}
+
+	if d.Truncate(multipleOf) != d {
+		return d, AliquotDurationError{"aliquot	duration error", multipleOf}
```

Comment:
Is that Msg `"aliquot	duration error"` (with tab character in the middle, in fact) is used anywhere? It looks like those errors are custom formatted everywhere, so we can drop `Msg` field?

Reply:
(and we really need a replacement for "aliquot")

Reply:
`NotMultipleOfError`?

Reply:
"Fractional"? But anything is better than "aliquot" – every developer will have to use a dictionary for that word 😆